docs/src/man/convex_sets.md

```diff
@@ -0,0 +1,269 @@
+plot!(xlims=(0., 2.), ylims=(0., 2.), xticks=[0., 1., 2.],  # hide
+      yticks=[1., 2.], top_margin=3mm)  # hide
+plot1(E, true)  # hide
+```
+
+### Singleton
+
+The next simplest convex set is the singleton, which consists of a single point.
+
+```@example convex_sets
+S = Singleton([1.0, 1.0])
+
+plot_raw()  # hide
+plot!(S, markercolor=c2, markersize=10, xlims=(0., 2.), ylims=(0., 2.),  # hide
+      xticks=[0., 1., 2.], yticks=[1., 2.], top_margin=3mm)  # hide
+```
+
+### Unit balls
+
+Now we consider sets that represent the unit balls in different ``p``-norms.
+
+```math
+    ‖ x = (x_1, …, x_n) ‖_p := \sqrt[^p]{|x_1|^p + … + |x_n|^p}
+```
+
+Formally, a unit ball is the sets of points ``x`` with ``‖x‖_p ≤ 1``.
+
+Balls for ``p ≥ 1`` are convex.
+
+#### Infinity norm
+
+The unit ball in the infinity norm (``p = ∞``) is also called *hypercube*.
+
+```@example convex_sets
+Bi = BallInf([0.0, 0.0], 1.0)
+
+plot_raw()  # hide
+plot!(xlims=(-1., 1.), ylims=(-1., 1.), xticks=[-1., 0., 1.],  # hide
+      yticks=[0., 1.], top_margin=3mm)  # hide
+plot1(Bi, true)  # hide
+```
+
+#### Euclidean norm
+
+The unit ball in the Euclidean norm (``p = 2``) is also called *hypersphere*.
+
+```@example convex_sets
+B2 = Ball2([0.0, 0.0], 1.0)
+
+plot_raw()  # hide
+plot!(xlims=(-1., 1.), ylims=(-1., 1.), xticks=[-1., 0., 1.],  # hide
+      yticks=[0., 1.], top_margin=3mm)  # hide
+plot2(B2, true)  # hide
+```
+
+#### Manhattan norm
+
+The unit ball in the Manhattan norm (``p = 1``) is also called *cross-polytope*.
+
+```@example convex_sets
+B1 = Ball1([0.0, 0.0], 1.0)
+
+plot_raw()  # hide
+plot!(xlims=(-1., 1.), ylims=(-1., 1.), xticks=[-1., 0., 1.],  # hide
+      yticks=[0., 1.], top_margin=3mm)  # hide
+plot3(B1, true)  # hide
+```
+
+#### More unit balls
+
+Below we show more exotic example of unit balls for ``p = 3, 42, π``.
+
+```@example convex_sets
+B3 = Ballp(3., [0.0, 0.0], 1.0)
+
+plot_raw()  # hide
+plot!(xlims=[-1., 1.], ylims=[-1., 1.], xticks=[-1., 0., 1.],  # hide
+      yticks=[0., 1.])  # hide
+plot1(B3, 1e-5, true)  # hide
+```
+
+```@example convex_sets
+B42 = Ballp(42., [0.0, 0.0], 1.0)
+
+plot_raw()  # hide
+plot!(xlims=[-1., 1.], ylims=[-1., 1.], xticks=[-1., 0., 1.],  # hide
+      yticks=[0., 1.])  # hide
+plot2(B42, 1e-5, true)  # hide
+```
+
+```@example convex_sets
+Bπ = Ballp(π - 2., [0.0, 0.0], 1.0)
+
+plot_raw()  # hide
+plot!(xlims=[-1., 1.], ylims=[-1., 1.], xticks=[-1., 0., 1.],  # hide
+      yticks=[0., 1.])  # hide
+plot3(Bπ, 1e-5, true)  # hide
+```
+
+Finally, the following example of the unit ball for ``p = 2/3`` demonstrates
+that for ``p < 1`` the unit balls are not convex.
+
+![../assets/ball_two_thirds.png](../assets/ball_two_thirds.png)
+
+[Source: [Wikimedia](https://commons.wikimedia.org/wiki/File:Astroid.svg)]
+
+### Unbounded sets
+
+#### Hyperplane
+
+A hyperplane is the generalization of the line (in 2D) and the plane (in 3D).
+Formally, a hyperplane is parametric in a normal vector ``d`` and a constant
```

Comment:
(i would use letters `a` and `b` because we use that in other parts of the documentation and the code)